radiomics/imageoperations.py

```diff
@@ -11,6 +11,42 @@
 logger = logging.getLogger(__name__)
 
 
+def getMask(mask, **kwargs):
+  """
+  Function to get the correct mask. Includes enforcing a correct pixel data type.
+
+  Also supports extracting the mask for a segmentation (stored as SimpleITK Vector image) if necessary.
+  In this case, the mask at index ``label - 1`` is extracted. All non-zero pixels in that mask are then set to the
+  value of the label.
+
+  :param mask: SimpleITK Image object representing the mask. Can be a vector image to allow for overlapping masks.
```

Comment:
Do I understand it correctly that `label` argument has dual meaning - to set the value of the label that should be used in the mask, when single volume, and to set the vector component otherwise? If this is the case, I think this is confusing. I would introduce additional argument to select the vector component.

Reply:
Correct, although I considered it more from the viewpoint of Slicer's Segmentation/Labelmap objects, where ROIs are defined either by their index in the vector image (Segmentation, supports overlap) __or__ the pixel value (Labelmap, does not support overlap).

In this setting, the `label` parameter just defines which ROI you want, regardless of the method you used to store it (Segmentation or Labelmap).

Reply:
Yes, I see. The issue is, there is probably no way to know how the input vector image was created, and whether this assumption holds. It's your call!

Reply:
I think in practice, either way is probably fine, since I don't think this format is used widely. I just think that it is confusing to overload parameters and give them implicit meaning in specific context. We also don't know whether conventions used currently in Slicer Segmentations will not change over time, or whether Slicer developers can write code that would create programmatically `.seg.nrrd` that does not follow this convention. Overall, I think it is cleaner to separate parameters for channel selection and for the label within channel. We could also have a default setting that channel is set to be identical to label if not specified, but leave the flexibility to adjust the channel if needed. But this is just my thoughts. I am fine if you proceed with your current implementation.

Reply:
It's a good point that the '.seg.nrrd' convention is ad hoc and could change without notice.  We should at least suggest that there be a version number added.  I'm not even sure it's documented anywhere.

Having an independent flag for channel selection and label within channel also makes sense.  Then a shorthand 'seg.nrrd' mode would set the values according to segment editor conventions.

Throwing exceptions or at least warnings for any unexpected input would also be good (e.g. a vector image that isn't a seg.nrrd isn't likely to make sense as a label definition).